eox_nelp/pearson_vue/api/v1/views.py

```diff
@@ -0,0 +1,139 @@
+from rest_framework.response import Response
+
+from eox_nelp.pearson_vue.api.v1.serializers import PearsonRTENSerializer
+from eox_nelp.pearson_vue.models import PearsonRTENModel
+
+
+class PearsonRTENBaseView(generics.CreateAPIView):
+    """
+    Base view for Pearson RTEN (Real Time Event Notification) API endpoints.
+
+    This view provides the base functionality for creating Pearson RTEN events.
+    Subclasses should define the `event_type` attribute to specify the type of event.
+
+    Attributes:
+        permission_classes (list): List of permission classes required for the view.
+        serializer_class (Serializer): Serializer class used for validating and parsing data.
+        queryset (QuerySet): Queryset representing the model instances to operate on.
+        authentication_classes (list): List of authentication classes used for authentication.
+    """
+    event_type = None
+    permission_classes = [IsAuthenticated]
+    serializer_class = PearsonRTENSerializer
+    queryset = PearsonRTENModel.objects.all()  # pylint: disable=no-member
+    authentication_classes = [JwtAuthentication]
+
+    def perform_create(self, serializer):
+        """
+        Perform creation of Pearson RTEN event.
+
+        This method is called when creating a new Pearson RTEN event.
+        It extracts the content data from the request and saves the event using the provided serializer.
+
+        Args:
+            serializer (Serializer): The serializer instance used for data validation and saving.
+        """
+        content_data = self.request.data.copy()
+        serializer.save(event_type=self.event_type, content=content_data)
+
+    def create(self, request, *args, **kwargs):
+        """
+        Create a new Pearson RTEN event.
+
+        Overrides the default create method to return a response with status code 201
+        and an empty dictionary.
+
+        Args:
+            request (Request): The request object containing the data.
+
+        Returns:
+            Response: Response object with status code 201 and an empty dictionary.
+        """
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+        headers = self.get_success_headers(serializer.data)
+        return Response({}, status=status.HTTP_200_OK, headers=headers)
+
+
+class ResultNotificationView(PearsonRTENBaseView):
+    """
+    View for handling Result Notification events.
+
+    This view handles the creation of Result Notification events in the Pearson RTEN system.
+    The `event_type` attribute is set to "resultNotification".
+    """
+
+    event_type = "resultNotification"
+
+
+class PlaceHoldView(PearsonRTENBaseView):
+    """
+    View for handling Place Hold events.
+
+    This view handles the creation of Place Hold events in the Pearson RTEN system.
+    The `event_type` attribute is set to "placeHold".
+    """
+
+    event_type = "placeHold"
```

Comment:
Could we create something like constant and load for that something like 
https://github.com/eduNEXT/eox-nelp/pull/164/files#diff-c9d49fedc8ccd5477e993f68a6696079ac46a3fbceb64f6f7e3af9c8f5d4ccf1R15
I mean also the test use the event type, and seem has the same shape in URLs to give the name.

Reply:
Changed in the constants implementation https://github.com/eduNEXT/eox-nelp/pull/164/commits/98fb31ce2d16c6af30c5a8708e579d432f4d272b